src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/PseudoCustomAttributesData.cs

```diff
@@ -139,17 +139,14 @@ internal BlobBuilder SerializeMarshallingData()
                 case UnmanagedType.CustomMarshaler:
                     writer.WriteUInt16(0); // padding
 
-                    switch (_marshalTypeNameOrSymbol)
+                    if (_marshalTypeName != null)
```

Comment:
So the Type case no longer needed and we just can use a string? -- it appears it was previously used for arrays.

Reply:
Yep, the `CustomAttibutesBuilder` emits the type name when populating the field value https://github.com/dotnet/runtime/blob/84bc85c566cb3e924a5cb3c496a83e76a1c35f08/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/CustomAttributeBuilder.cs#L426-L436

Initially it was trying to load the type from the full name when deserializing, just to write the type's full name back when serializing it to file. Now we just keep the full name as is and serialize it back to file as is. We had discussion about this above https://github.com/dotnet/runtime/pull/95753#discussion_r1420874040

The array case is also updated below.